packages/playwright-core/src/server/injected/clock.ts

```diff
@@ -282,10 +285,12 @@ export class ClockController {
 
     this._advanceNow(timer.callAt);
 
-    if (timer.type === TimerType.Interval)
-      this._timers.get(timer.id)!.callAt += timer.delay;
-    else
+    if (timer.type === TimerType.Interval) {
+      const callAt = this._timers.get(timer.id)!.callAt;
```

Comment:
I wonder whether `this._timers.get(timer.id) === timer`?

Reply:
I'm being stupid - doing a local refactoring w/o looking at context.